README.md

```diff
@@ -1 +1,117 @@
+                        directives: |
+                          {
+                            "waf1":{
+                                  "simple_directives":[
+                                          "SecDefaultAction \"phase:3,log,auditlog,pass\"",
+                                          "SecDefaultAction \"phase:4,log,auditlog,pass\"",
+                                          "SecDefaultAction \"phase:5,log,auditlog,pass\"",
+                                          "SecDebugLogLevel 3",
+                                          "SecRule REQUEST_URI \"@streq /admin\" \"id:101,phase:1,t:lowercase,deny\" \nSecRule REQUEST_BODY \"@rx maliciouspayload\" \"id:102,phase:2,t:lowercase,deny\" \nSecRule RESPONSE_HEADERS::status \"@rx 406\" \"id:103,phase:3,t:lowercase,deny\" \nSecRule RESPONSE_BODY \"@contains responsebodycode\" \"id:104,phase:4,t:lowercase,deny\""
+                                    ],
+                                  "directives_files":[
+                                          "/etc/envoy/rules/coraza-demo.conf",
+                                          "/etc/envoy/rules/crs-setup-demo.conf", 
+                                          "/etc/envoy/rules/crs/*.conf"
+                                    ]
+                              }
+                          }
+                        default_directive: "waf1"
+                        host_directive_map: |
+                          {
+                            "foo.example.com":"waf1",
+                            "bar.example.com":"waf1"
+                          }
+```
+
+### Using CRS
+
+you can use the [Core Rule Set](https://github.com/coreruleset/coreruleset) by mounting it to the container or you can also mount other rules
+
+
+#### Recommendations using CRS with proxy-wasm
+
+- In order to mitigate as much as possible malicious requests (or connections open) sent upstream, it is recommended to keep the [CRS Early Blocking](https://coreruleset.org/20220302/the-case-for-early-blocking/) feature enabled (SecAction [`900120`](./wasmplugin/rules/crs-setup.conf.example)).
+
+### Running go-ftw (CRS Regression tests)
+
+The following command runs the [go-ftw](https://github.com/coreruleset/go-ftw) test suite against the filter with the CRS fully loaded.
+
+```bash
+go run mage.go ftw
+```
+
+Take a look at its config file [ftw.yml](./ftw/ftw.yml) for details about tests currently excluded.
+
+One can also run a single test by executing:
+
+```bash
+FTW_INCLUDE=920410 go run mage.go ftw
+```
+
+### Use Compare
+If you want to compare the performance of two plugins, just follow these steps
+1. Pull the Envoy docker image 
+```bash
+docker pull envoyproxy/envoy: contrib-dev
+```
+2. Set up envoy's upstream server
+
+To do this, we simply need an http server listening on our local machine on port 8081 and providing a get /ping interface
+
+3. Run the compare/compare. Go to (want to test alone Envoy go or delete other methods wasm)
```

Comment:
Sorry, I do not understand `Go to (want to test alone Envoy go or delete other methods wasm)`. What does it mean?

Reply:
With two test function in the example below, if just want to test one delete another can, do not need other operations

Reply:
I see. Let's add the explanation to the doc.
BTW, is it possible to set up the upstream on 8081 by the test suite, via docker-compose or other similar methods?

Reply:
You can switch to any upstream server by changing the envoy-config.yaml in envoy_go directory or the wasm directory

Reply:
As long as the network is reachable